docs/shared/react-tutorial/4-workspace-optimization.md

```diff
@@ -0,0 +1,231 @@
+        "assets": ["libs/products/*.md"]
+      }
+    },
+    "lint": {
+      "executor": "@nrwl/linter:eslint",
+      "outputs": ["{options.outputFile}"],
+      "options": {
+        "lintFilePatterns": ["libs/products/**/*.ts"]
+      }
+    },
+    "test": {
+      "executor": "@nrwl/jest:jest",
+      "outputs": ["coverage/libs/products"],
+      "options": {
+        "jestConfig": "libs/products/jest.config.ts",
+        "passWithNoTests": true
+      }
+    }
+  },
+  "tags": []
+}
+```
+
+Outputs are stored in the cache so that terminal output can be replayed, and any created files can be pulled from your cache, and placed where they were created the original time the task was run.
+
+{% callout type="note" title="Distributed Caching" %}
+Without using Nx Cloud, your cache is only on your local filesystem (you can see it in your `node_modules/.cache` directory of your workspace).
+
+However if you are using Nx Cloud, running tasks will also populate the centralized cache by sharing your runs with Nx Cloud. This can be particularly helpful for not duplicating tasks between you and your CI processes, as well as for caching tasks from other teammates.
+
+You can read more about Nx Cloud in [the Nx Cloud introduction](/nx-cloud/intro/what-is-nx-cloud).
+{% /callout %}
+
+### Example
+
+To see caching in action, run the command `npx nx build admin`:
+
+```bash
+> npx nx build admin
+
+> nx run admin:build:production
+
+Entrypoint main 139 KiB = runtime.54e36ebee261465d.js 1.19 KiB main.623d91691bdb6754.css 42 bytes main.303fe7c1dcf5306b.js 137 KiB
+Entrypoint polyfills 93.5 KiB = runtime.54e36ebee261465d.js 1.19 KiB polyfills.bd0d0abec287a28e.js 92.3 KiB
+Entrypoint styles 1.19 KiB = runtime.54e36ebee261465d.js 1.19 KiB styles.ef46db3751d8e999.css 0 bytes
+chunk (runtime: runtime) main.623d91691bdb6754.css, main.303fe7c1dcf5306b.js (main) 144 KiB (javascript) 50 bytes (css/mini-extract) [initial] [rendered]
+chunk (runtime: runtime) polyfills.bd0d0abec287a28e.js (polyfills) 301 KiB [initial] [rendered]
+chunk (runtime: runtime) styles.ef46db3751d8e999.css (styles) 50 bytes (javascript) 80 bytes (css/mini-extract) [initial] [rendered]
+chunk (runtime: runtime) runtime.54e36ebee261465d.js (runtime) 3.23 KiB [entry] [rendered]
+webpack compiled successfully (0c0df3e6c70c6b7b)
+
+ ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target build for project admin (4s)
+```
+
+Since you had not run the `build` target before for the `admin` project, Nx ran the `build`, and populated the results in `dist/apps/admin` as specified in the `admin` project's `project.json` file for the `build` target, which we can see took 4 seconds.
```

Comment:
```suggestion
Since you have not run the `build` target before for the `admin` project, Nx runs the `build`, and populates the results in `dist/apps/admin` as specified in the `admin` project's `project.json` file for the `build` target, which we can see takes 4 seconds.
```